yt_dlp/extractor/substack.py

```diff
@@ -50,7 +50,7 @@ def _extract_embed_urls(cls, url, webpage):
         if not re.search(r'<script[^>]+src=["\']https://substackcdn.com/[^"\']+\.js', webpage):
             return
 
-        mobj = re.search(r'{[^}]*["\']subdomain["\']\s*:\s*["\'](?P<subdomain>[^"]+)', webpage)
+        mobj = re.search(r'{[^}]*\\?["\']subdomain\\?["\']\s*:\s*\\?["\'](?P<subdomain>[^\\"]+)', webpage)
```

Comment:
Yikes, should've been doing this the whole time
```suggestion
        mobj = re.search(r'{[^}]*\\?["\']subdomain\\?["\']\s*:\s*\\?["\'](?P<subdomain>[^\\"\']+)', webpage)
```

Reply:
Sorry, but what's the single quote for? JSON requires double quotes.

Reply:
Based on the already existing pattern, it would be to close the opening `["\']`

But if you say we will always be searching valid JSON with double-quotes, then all of the single-quotes are unnecessary?

Reply:
Oops, sorry, haven't noticed them already being present in the pattern.